sympy/geometry/point.py

```diff
@@ -527,45 +628,37 @@ def __add__(self, other):
         sympy.geometry.entity.translate
 
         """
-
-        if iterable(other):
+        if not isinstance(other, (GeometryEntity, Point)):
+            other = Point.pointify(other,
+                dimension=self.ambient_dimension, ensure_point=False)
+        if isinstance(other, (Point, Tuple)):
```

Comment:
Seeing this code makes me nervous.  Perhaps a comment saying that `pointify` will return a `GeometricEntity` or a `Tuple`?